realm/src/androidTest/java/io/realm/RealmQueryTest.java

```diff
@@ -36,19 +38,40 @@ protected void tearDown() throws Exception {
             testRealm.close();
     }
 
-    private void populateTestRealm(int objects) {
+    // Populate testing data for query testing which will be used for AllTypes and AllTypesIndex.
+    // Param clazz only takes AllTypes or AllTypesIndex for now
+    // So please keep:
+    // 1. AllTypesIndex always has the same field names, getters and setters with AllTypes
+    // 2. Use reflection to set/get data
+    // TODO: Switch to DynamicRealmObject when it is ready
+    private void populateTestRealm(int objects, Class clazz) {
```

Comment:
I would be a bit hesitant about adding this level of reflection to all test methods in this class. Couldn't you instead make an interface that both AllTypes and AllTypesIndex implement, then you could do the following (which would be both faster and more type safe): 

```
    private void populateTestRealm(int objects, Class<? extends RealmObject> clazz) {
        testRealm.beginTransaction();
        testRealm.allObjects(AllTypes.class).clear();
        testRealm.allObjects(NonLatinFieldNames.class).clear();
        for (int i = 0; i < objects; ++i) {
            AllTypesInterface allTypes = (AllTypesInterface) testRealm.createObject(clazz);
```